src/java.base/share/classes/jdk/internal/loader/URLClassPath.java

```diff
@@ -439,34 +439,56 @@ private synchronized Loader getLoader(int index) {
                 continue;
             }
             // Otherwise, create a new Loader for the URL.
-            Loader loader;
+            Loader loader = null;
+            final URL[] loaderClassPathURLs;
             try {
                 loader = getLoader(url);
-                // If the loader defines a local class path then add the
-                // URLs as the next URLs to be opened.
-                URL[] urls = loader.getClassPath();
-                if (urls != null) {
-                    push(urls);
-                }
+                // If the loader defines a local class path then construct
+                // and add the URLs as the next URLs to be opened.
```

Comment:
The change to URLClassLoad looks okay although I think the original comment was a bit clearer, no need to insert "then construct" here as this is just getting the class path URLs.

Reply:
Hello Alan, I've reverted the change to this code comment in the latest update to this PR.